### PR TITLE
Create Connect Activity

### LINKED ISSUE #39: Create Bootup Activity

### PR DESCRIPTION
**Pull request by jennaluz**
This activity will handle the layout of the display upon user connection and disconnection to the access point.

### CODE REVIEW

File: src/lumen/activity/connect.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+namespace lumen::activity {
+namespace {
+
+constexpr auto tag = "activity/connect";
+
+}
+
+Connect::Connect()
+    : ssid_{reinterpret_cast<char const*>(CONFIG_NET_WIFI_SSID)},
```
Comment (loganfin) [inline src/lumen/activity/connect.cpp]: Remove the `reinterpret_cast`s here and initialize these members in the class. Then you should be able to remove this constructor.

File: src/lumen/activity/connect.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      mdns_hostname_{reinterpret_cast<char const*>(CONFIG_NET_MDNS_HOSTNAME)}
+{
+}
+
+void Connect::set_connected(bool connected)
+{
+    ESP_LOGI(tag, "Setting up connection %d", connected);
```
Comment (loganfin) [inline src/lumen/activity/connect.cpp]: Remove this debug statement.

File: src/lumen/activity/connect.cpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    connected_ = connected;
+    update_display();
+}
+
+void Connect::update_display()
+{
+    auto* display = get_display();
+
+    display->clearScreen();
+    display->setTextColor(hardware::Display::color565(255, 128, 0));
```
Comment (loganfin) [inline src/lumen/activity/connect.cpp:29]: Not important now, but in the future we might want to have some theme-specific color values so we can do something like:

```
display->setTextColor(hardware::Display::Color::text);
```

File: src/lumen/activity/connect.hpp [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+#include "lumen/activity/field/text.hpp"
+
+namespace lumen::activity {
+
+/** Stores and displays the data for a user to connect to the board.
+ * This activity should be active when no user is connected.
+ */
+class Connect : public Activity {
+public:
+    Connect();
```
Comment (loganfin) [inline src/lumen/activity/connect.hpp]: Remove the constructor and initialize the member variables here.

File: src/lumen/button_callback.hpp [hunk 5]
```diff
@@ -0,0 +1,7 @@
+#pragma once
+
+namespace lumen {
+
```
Comment (jennaluz) [inline src/lumen/button_callback.hpp:4]: - [x] Add comment

File: src/lumen/net/wifi.cpp [hunk 6]
```diff
@@ -1,18 +1,37 @@
...
+constexpr auto esp_max_password_length = 64;
+
+wifi_config_t config{.ap{
+    .ssid = CONFIG_NET_WIFI_SSID,
+    .password = CONFIG_NET_WIFI_PASSWORD,
+    .ssid_len = strlen(CONFIG_NET_WIFI_SSID),
+    .authmode = WIFI_AUTH_WPA_WPA2_PSK,
+    .max_connection = CONFIG_NET_WIFI_MAX_STA_CONNECTION,
+}};
+
```
Comment (jennaluz) [inline src/lumen/net/wifi.cpp:30]: - [x] Add comments

File: src/lumen/net/wifi.cpp [hunk 7]
```diff
@@ -30,41 +49,70 @@ void wifi_event_handler(
...
+{
+    randomize_wifi_password();
+    log_wifi_credentials();
+}
+
+namespace {
+
+void randomize_wifi_password()
+{
+    ESP_LOGI(tag, "Randomize softAP password");
```
Comment (loganfin) [inline src/lumen/net/wifi.cpp]: I don't think we need this.

File: src/lumen/net/wifi.cpp [hunk 8]
```diff
@@ -30,41 +49,70 @@ void wifi_event_handler(
...
+    std::srand(time(0));
+    for (int i = 0; i < length; i++) {
+        password[i] = static_cast<uint8_t>(rand() % 26 + 97);
+    }
+
+    std::memcpy(config.ap.password, password, esp_max_password_length);
+
+    ESP_ERROR_CHECK(esp_wifi_set_config(WIFI_IF_AP, &config));
+
+    ESP_LOGI(tag, "New password: %s", password);
```
Comment (loganfin) [inline src/lumen/net/wifi.cpp]: I don't think we need this either.

File: src/lumen/net/wifi.hpp [hunk 9]
```diff
@@ -1,8 +1,13 @@
...
 
+#include "lumen/activity/context.hpp"
+
+#include <string>
+
 namespace lumen::net {
 
-/// Initialize the ESP32 in softAP mode.
-void init_wifi_softap();
+void init_wifi(activity::Context& activity_context);
```
Comment (loganfin) [inline src/lumen/net/wifi.hpp:13]: Add some comments here.

**Review by loganfin — CHANGES_REQUESTED** [review]

File: src/lumen/button_callback.hpp [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+/** The handler for a long press of the power button.
+ *
+ * \param button The handle to the button.
```
Comment (loganfin) [inline src/lumen/button_callback.hpp:7]: Is this accurate?

**Review by loganfin — CHANGES_REQUESTED** [review]

**Review by loganfin — APPROVED** [review]

**Pull request merged**